### PR TITLE
feat: add backupBackingImage table in backup page with tabs

### PR DESCRIPTION
**Pull request by houhoucoop**
### What this PR does / why we need it
- [x] using [tabs](https://ant.design/components/tabs) to separate volume backup and backing image backup
- [x] Introduce the `BackupVolume` and `BackupBackingImage` component to enhance code reusability
- [x] The tab can be accessed via its URL

### Issue
[[IMPROVEMENT][UI] Add backupBackingImage table in backup page with tabs #8956
](https://github.com/longhorn/longhorn/issues/8956)

### Test Result
- Navigate to the `Backup` page
- Verify the page displays tabs labeled `Volume` and `Backing Image`
- Confirm the default tab is `Volume`, and the URL updates to `/#/backup#volume`
Screenshot 2024-12-12 at 2 54 16 PM (2)
- Clicking the `Backing Image` tab should display the table correctly and update the URL to `/#/backup#backing-image`
Screenshot 2024-12-12 at 2 53 20 PM (2)
- Accessing the page directly via `http://localhost:8080/#/backup#backing-image` should automatically activate the `Backing Image` tab
- The Backing Image page should function as usual
Screenshot 2024-12-12 at 3 01 33 PM (2)

### Additional documentation or context
Test with `LONGHORN_MANAGER_IP=http://139.59.122.79:30001/ npm run dev`

## Release Notes

- **New Features**
	- Introduced a new `BackupBackingImage` component for managing backup images.
	- Added a `BackupVolume` component for managing backup volumes with enhanced functionalities.
	- Implemented a tabbed interface in the `Backup` component for better navigation between "Volume" and "Backing Image".

- **Bug Fixes**
	- Adjusted column widths in the backup image list for improved data display.

- **Style**
	- Updated CSS styles for various components, including new container classes for layout enhancements.

- **Documentation**
	- Updated prop types and export statements for several components to reflect new naming conventions and functionalities.

### CODE REVIEW

File: src/routes/backup/VolumeBackup.js [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+VolumeBackup.propTypes = {
+  backup: PropTypes.object,
+  location: PropTypes.object,
+  dispatch: PropTypes.func,
+  loading: PropTypes.bool,
+  setting: PropTypes.object,
+  backingImage: PropTypes.object,
+}
+
+export default connect(({ backup, setting, backingImage, loading }) => ({ backup, setting, backingImage, loading: loading.models.backup }))(VolumeBackup)
```
Comment (houhoucoop) [inline src/routes/backup/VolumeBackup.js:349]: The code has been moved from the `backup/index.js`, and the logic remains unchanged.
Ref: https://github.com/longhorn/longhorn-ui/pull/827/files#diff-e5e7e3c3e107db474c456a6d74c30dbe17e72b2668a86702c0cdfb7ce7b42354L19

File: src/routes/backingImage/BackupBackingImage.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      }
+    }
+
+    calculateTableBodyHeight()
+    window.addEventListener('resize', calculateTableBodyHeight)
+
+    return () => {
+      window.removeEventListener('resize', calculateTableBodyHeight)
+    }
+  }, [])
```
Comment (houhoucoop) [inline src/routes/backingImage/BackupBackingImage.js:38]: Only the 1useEffect1 for calculating the table body height was added, the rest of the logic is moved from `backingImage/index.js`.

File: src/routes/backingImage/utils.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      case 'name':
+      case 'state':
+      case 'url':
+        result = bbiData.filter((image) => (value ? image[field].toLowerCase().includes(value.toLowerCase().trim()) : true))
+        break
+      default:
+    }
+  }
+  return result
+}
```
Comment (houhoucoop) [inline src/routes/backingImage/utils.js:40]: Moved from `backingImage/index.js`, no changes to the logic.

File: src/routes/backup/BackupVolume.js [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+BackupVolume.propTypes = {
+  backup: PropTypes.object,
+  location: PropTypes.object,
+  dispatch: PropTypes.func,
+  loading: PropTypes.bool,
+  setting: PropTypes.object,
+  backingImage: PropTypes.object,
+}
+
+export default connect(({ backup, setting, backingImage, loading }) => ({ backup, setting, backingImage, loading: loading.models.backup }))(BackupVolume)
```
Comment (houhoucoop) [inline src/routes/backup/BackupVolume.js:349]: Moved from the `backup/index.js`, no changes to the logic.
Ref: https://github.com/longhorn/longhorn-ui/pull/827/files#diff-e5e7e3c3e107db474c456a6d74c30dbe17e72b2668a86702c0cdfb7ce7b42354L19

**Review by derekbit — APPROVED** [review]
LGTM

**Comment by derekbit** [comment]
cc @ChanYiLin

File: src/routes/backup/BackupVolume.js [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+      })
+    },
+    Create(record) { // to create DR volume
+      dispatch({
+        type: 'backup/CreateStandVolume',
+        payload: record,
+      })
+    },
+    DeleteAllBackups(record) {
+      // dispatch({
```
Comment (a110605) [inline src/routes/backup/BackupVolume.js:79]: comments can be removed if not requried.

**Review by ChanYiLin — APPROVED** [review]
LGTM
Thanks

**Pull request merged**